---
title: Warfarin Pharmacomicrobiomics
author: Heidi Steiner
date: '2020-07-08'
slug: warfarin-pharmacomicrobiomics
categories: []
tags:
  - 16S
  - warfarin
subtitle: ''
summary: ''
authors: []
lastmod: '2020-07-08T14:57:38-07:00'
featured: no
image:
  caption: ''
  focal_point: ''
  preview_only: no
projects: []
---



<p>Warfarin pharmacogenomics has been on my mind for quite awhile. Warfarin acts as an anticoagulant by inhibiting the vitamin K cycle which is responsible for creating blood clotting factors. When I learned that vitamin K was produced by bacteria in the gut microbiome, I hypothesized that vitamin K producing capability of the gut microbiome would be associated with warfarin dose requirements. In my dissertation research I will investigate that exact question. Until we get that result, however, I will keep updated with the literature around the topic.</p>
<div id="the-study" class="section level2">
<h2>The study</h2>
<p>A paper titled “The gut microbes, Enterococcus and Escherichia-Shigella, affect the responses of heart valve replacement patients to the anticoagulant warfarin” was recently published suggesting the relationship between the gut microbiome and warfarin response. The study sequenced the 16S gene and the metagenome of stool samples of 80 and 15 warfarin users, respectively. The authors factorized the outcome by creating warfarin responder phenotype groups that were based on residuals from their previously published dose prediction model.</p>
</div>
<div id="the-results" class="section level2">
<h2>The results</h2>
<div id="s" class="section level4">
<h4>16S</h4>
<p>The bacterial genera Enterococcus, Bacteroides, Escherichia-Shigella, Klebsiella, Streptococcus, Lactobacillus, Prevotella_9, Bifidobacterium and Veillonella were the top 9 dominant taxa among all responder phenotypes. Only two genera, <em>Enterococcus</em> and <em>Escherichia-Shiigella</em> were abundantly different among the phenotype groups. <em>Enterococcus</em> proportions were higher in the patients who were overpredicted by the dosing algorithm. Proportions of <em>Escherichia-Shigella</em> were higher in underpredicted patients.</p>
<p><img src="fig3.jpg" /></p>
</div>
<div id="metagenomics" class="section level4">
<h4>Metagenomics</h4>
<p>The authors annotated genes using the KEGG functional database to investigate menaquinone biiosynthesis pathways in the groups. Interestingly, the patients underpredicted by the dosing algorithm contained increased levels of menaaquinone biosynthesis. The reverse was found in the opposite phenotype group.</p>
<p><img src="fig4.jpg" /></p>
</div>
</div>
<div id="the-conclusions" class="section level2">
<h2>The conclusions</h2>
<p>The gut microbiota may affect responses to anticoagulation therapy, specifically with warfarin. Many questions remain regarding this novel finding including 1) do these findings replicate, 2) what species are differently abundant, 3) what genes are upregulated, and 4) can this data be used to predict patient’s dose?</p>
</div>
