---
title: "qiime2R"
author: "Heidi Steiner"
date: 2020-05-06
categories: ["R"]
tags: ["R Markdown", "plot", "16S"]
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(collapse = TRUE)
library(qiime2R)
library(viridis)
library(tidyverse)
```

# Qiime
My sequencing was done by Ion Torrent which made Qiime a bit difficult to use because of lacking documentation. I followed, mostly, this post by Carli Jones: <https://forum.qiime2.org/t/possible-analysis-pipeline-for-ion-torrent-16s-metagenomics-kit-data-in-qiime2/13476>.

## Pipeline
1. Import fastq using "manifest" file format
2. Perform ```DADA2``` 
3. Explore ```feature counts```
4. Create phylogenetic tree using ```SEPP (fragment insertion)```
5. ```Alpha rarefaction``` 
6. ```Core metrics analysis``` 


# Qiime outputs, R plots

For this project data visualization is the most important part because it's pilot data for a new project. qiime2R has really good documentation; for details see <https://github.com/jbisanz/qiime2R>.

First I uploaded the  Qiime2 output files needed to make my plot which was very straightforward following the documentation by Dr. Jordan Bisanz.  

```{r warning=FALSE, message=F}
metadata<-read_tsv("warfgi_qiime_meta.tsv")
colnames(metadata)[colnames(metadata) == "sampleid"] = "SampleID"
uwunifrac<-read_qza("unweighted_unifrac_pcoa_results.qza")
shannon<-read_qza("shannon_vector.qza")$data %>% rownames_to_column("SampleID")
```

# Plotting

Again, using code from the qiime2r documentation, I plotted the first two PCs with sizing by Alpha Diversity. 

```{r warning=FALSE, message=F}
uwunifrac$data$Vectors %>%
  select(SampleID, PC1, PC2) %>%
  left_join(metadata) %>%
  left_join(shannon) %>%
  ggplot(aes(x=PC1, y=PC2, color=`samplegroup`, size=shannon)) +
  geom_point(alpha=0.8) + #alpha controls transparency and helps when points are overlapping
  theme_q2r() +
  scale_size_continuous(name="Shannon Diversity") +
  scale_color_discrete(name="Patient") + 
  scale_color_viridis(discrete = T, name = "patient")
```



# Tada! 